codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -446,18 +450,28 @@ public static Http2Headers toHttp2Headers(HttpHeaders inHeaders, boolean validat
     public static void toHttp2Headers(HttpHeaders inHeaders, Http2Headers out) {
         Iterator<Entry<CharSequence, CharSequence>> iter = inHeaders.iteratorCharSequence();
         CharSequenceMap<AsciiString> connectionBlacklist =
-            toLowercaseMap(inHeaders.getAll(HttpHeaderNames.CONNECTION));
+            toLowercaseMap(inHeaders.getAll(CONNECTION));
         while (iter.hasNext()) {
             Entry<CharSequence, CharSequence> entry = iter.next();
             final AsciiString aName = AsciiString.of(entry.getKey()).toLowerCase();
             if (!HTTP_TO_HTTP2_HEADER_BLACKLIST.contains(aName) &&
                 !connectionBlacklist.contains(aName)) {
                 // https://tools.ietf.org/html/rfc7540#section-8.1.2.2 makes a special exception for TE
-                if (aName.contentEqualsIgnoreCase(HttpHeaderNames.TE)) {
-                    if (AsciiString.containsIgnoreCase(entry.getValue(), HttpHeaderValues.TRAILERS)) {
-                        out.add(HttpHeaderNames.TE, HttpHeaderValues.TRAILERS);
+                if (aName.contentEqualsIgnoreCase(TE)) {
+                    if (AsciiString.indexOf(entry.getValue(), ',', 0) == -1) {
+                        if (AsciiString.contentEqualsIgnoreCase(entry.getValue(), TRAILERS)) {
+                            out.add(TE, TRAILERS);
+                        }
+                    } else {
+                        List<CharSequence> teValues = StringUtil.unescapeCsvFields(entry.getValue());
+                        for (CharSequence teValue : teValues) {
+                            if (AsciiString.contentEqualsIgnoreCase(teValue, TRAILERS)) {
```

Comment:
trim?

Reply:
good question... @carl-mastrangelo @Scottmitch @ejona86 WDYT ?

Reply:
Lets trim just to be safe .. https://github.com/netty/netty/pull/7422/files#r152127214